src/migrer/core.clj

```diff
@@ -45,11 +45,89 @@
    [clojure.java.io :as io]
    [clojure.java.jdbc :as jdbc]
    [clojure.spec.alpha :as s]
-   [clojure.string :as str])
+   [clojure.string :as str]
+   [datascript.core :as d])
   (:import
-   [org.postgresql.util PSQLException]
+   [java.lang Exception]
    [java.security MessageDigest]))
 
+(def schema
+  {:migration.meta/id {:db/unique :db.unique/identity}
+   :migration.meta/version {:db/cardinality :db.cardinality/one}
+   :migration.meta/dependencies {:db/cardinality :db.cardinality/many
+                                 :db/valueType :db.type/ref}
+   :migration.meta/pre {:db/cardinality :db.cardinality/many}
+   :migration.meta/post {:db/cardinality :db.cardinality/many}
+   :migration.meta/run? {:db/cardinality :db.cardinality/one}
+   :migration.meta/type {:db/cardinality :db.cardinality/one
+                         :db/valueType :db.type/ref}
+   :migration.raw/filename {:db/cardinality :db.cardinality/one}
+   :migration.raw/dependencies {:db/cardinality :db.cardinality/many}
+   :migration.raw/sql {:db/cardinality :db.cardinality/one}})
+
+(def fact-rules
```

Comment:
Add should-run rule:

[(Should-run ?e) [?e :migration.meta/run? true]]
[(should-run ?e) (prior-to ?ep ?e) [?ep :migration.meta/run? true]]